### ISSUE TITLE
Detecting phoneme

### ISSUE THREAD

**Issue by Terrance82** [Cpinkston/Zipfian-Project#1]
Hi I was wondering how your program could detect phoneme? Did you use any forced alignment or purely extract out based on the frequency?

**Comment by Cpinkston** [comment]
Hey Terrance, 

The program is relying purely on the magnitude of the frequencies. To make it faster/easier to work with for my 2 week project I split the frequencies into about 10-15 buckets using a log scale and it worked well for the time that I had to produce results. 

If you were looking to improve on this project I would recommend breaking down the frequencies into more buckets and using some more advanced machine learning techniques to train the model (TensorFlow). 

The most difficult part for me was capturing a cleanly labeled training set. For my project I trained it purely on my voice. To make it more robust I would suggest having others "donate" their voice to the cause.

Best of luck,
Connor

**Comment by Terrance82** [comment]
Hi Conner thanks for replying. However I'm very new to this so I'm looking ways to learn specifically how speech recognition works. Currently I'm trying to think what are the ways that we could extract phonetic symbols from wav files.

**Comment by Cpinkston** [comment]
With any machine learning application it will be difficult to try and understand the specific features that cause a specific output. What I am doing with this application is not too different from an algorithm deciding if a picture is of a square or a circle. Where as with images you are combining the magnitude of pixels with sound you are combining the magnitude of frequencies. 

For this to be effective you need to have a large (and accurate) training set. Just as a child learns what a word water by hearing it when near a sink, in a bath or near a pond the training set needs to exposed to variations of the same sounds (different voices). But once the child learns the word water he cannot tell you which neurons are firing in his or her brain to make them think of that word.

You could look in to trying to manually program what different sounds will be but I would imagine you would always run in to different edge cases that would be difficult to tweak. The advantage you have with my approach is that you can have a human label the wav files by listening to them and then the algorithms take care of the rest. Once the model is trained it will be able to listen and extract those sounds.

Hope this is helpful. Let me know if I didn't interpret your question correctly.
Connor

**Comment by Cpinkston** [comment]
You are most likely looking for the create_data_set.py file to see how I transformed the sound in to the individual frequencies.

**Comment by Cpinkston** [comment]
After looking at your questions on another thread I think I have more context on what you are trying to do. It looks like you are trying to take a audio file and transcribe the words out in to a new file. The code that I have here will help you to train a model that can identify sounds as they are heard. You will most likely need to then train a model on that output to turn the phonetic sounds in to words. I would use something like a LSTM model.

**Comment by Terrance82** [comment]
Hi sorry for the late response, Thank you for the wonderful insights I will look into it and let you know how it goes.

**Comment by Terrance82** [comment]
Oh I forgot to mentioned that I'm trying to do something like pronunciation evaluation, where I get audio input and then display the phoneme / phones to see if the speaker's pronunciation is accurate.
I would use some Dictionary to see how the word is supposed to be pronounce.
